### PR TITLE
Fix(TDI-42424):Improve for tELTPostgresqlOutput generate wrong update

### PR DESCRIPTION
**Pull request by vyu-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-42424

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tELTPostgresqlOutput/tELTPostgresqlOutput_main.javajet [hunk 1]
```diff
@@ -110,271 +110,277 @@ if(useExistingConn) {
...
+
+	if(("INSERT").equals(dataAction)){
+%>	
+	String insertQuery_<%=cid %> = "INSERT INTO "+tableName_<%=cid%>+"(<%=insertColName.toString()%>)  ("+select_query_<%=cid %>+")";
+	pstmt_<%=cid %> = conn_<%=cid%>.prepareStatement(insertQuery_<%=cid %>);
+
+<%
+	}else if (("UPDATE").equals(dataAction)){
+%>
+	String updateQuery_<%=cid %> = "UPDATE "+tableName_<%=cid%>+" SET <%=updateSetStmt.toString()%> " +select_query_<%=cid %>.substring(select_query_<%=cid %>.indexOf("FROM"));
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTPostgresqlOutput/tELTPostgresqlOutput_main.javajet]: refer to https://github.com/Talend/tdi-studio-se/commit/7349dce3a880062811d0b8b5445d49677ba9f5ba

File: main/plugins/org.talend.designer.components.localprovider/components/tELTPostgresqlOutput/tELTPostgresqlOutput_main.javajet [hunk 2]
```diff
@@ -110,271 +110,277 @@ if(useExistingConn) {
...
+		if(colStmt.isAddCol()){
+			insertColName.append(colStmt.getName()+suffix);
+			insertValueStmt.append(colStmt.getSqlStmt()+suffix);
+			updateSetStmt.append(colStmt.getName());
+			updateSetStmt.append(colStmt.getSqlStmt()+suffix);
+		}else{
+			insertColName.append(colStmt.getName()+suffix);
+			insertValueStmt.append(colStmt.getValue()+suffix);
+			if(!colStmt.getColumn().isKey()){
+				updateSetStmt.append(colStmt.getName()+"=\"+routines.system.StringUtils.splitSQLColumns(selectQueryColumnsName_"+ cid +")[ "+ counterOuter + "]+\"" +suffix);
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTPostgresqlOutput/tELTPostgresqlOutput_main.javajet]: this done https://github.com/Talend/tdi-studio-se/commit/3a99f815264fa9fa9aedd888434d8500960f2637

File: main/plugins/org.talend.designer.components.localprovider/components/tELTPostgresqlOutput/tELTPostgresqlOutput_main.javajet [hunk 3]
```diff
@@ -110,271 +110,277 @@ if(useExistingConn) {
...
+	pstmt_<%=cid %> = conn_<%=cid%>.prepareStatement(insertQuery_<%=cid %>);
+
+<%
+	}else if (("UPDATE").equals(dataAction)){
+%>
+	String updateQuery_<%=cid %> = "UPDATE "+tableName_<%=cid%>+" SET <%=updateSetStmt.toString()%> " +select_query_<%=cid %>.substring(select_query_<%=cid %>.indexOf("FROM"));
+<%
+		if(CodeGenerationUtils.hasAlphaNumericCharacter(whereClause)) {
+%>
+		if(select_query_<%=cid %>.indexOf("WHERE")==-1){
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tELTPostgresqlOutput/tELTPostgresqlOutput_main.javajet]: this changes in greenplumn done in this commit 7349dce

**Comment by vyu-talend** [comment]
> i think the old fix on greenplumn feature not fix in postgres sql.
> so there are some difference between postgresql and greenplumn
> 
> This PR seems ok. but please do mre test about this.

Ok, I will check it and find out the differences.

**Pull request merged**